javaparser-core/src/main/java/com/github/javaparser/jctree/ast/visitor/APretty.java

```diff
@@ -0,0 +1,1172 @@
+package  com.github.javaparser.jctree.ast.visitor;
+
+import com.github.javaparser.jctree.ast.*;
+import com.sun.tools.javac.tree.*;
+import com.sun.tools.javac.tree.JCTree.*;
+import java.io.*;
+
+/* TODO - The JC add elements to the AST that are not AJC-
+* Thus, all methods should test for their coresponding AJC type before printing the comments,
+* So, a non-AJC element can be visited without raising and Exception, caused by bad type-casting.
+* Other option: create a visitor that checks if there are a non-AJC node, and convert it.
+*/
+public class APretty extends Pretty {
+   public APretty (Writer out, boolean sourceOutput) {
+      super(out,sourceOutput);
+   }
+   public void visitTopLevel(JCCompilationUnit that) {
+      if (((AJCCompilationUnit) that).hasComment()) {
+         try {
+            println();
+            print("/*" + ((AJCCompilationUnit) that).getComment() + "*/");
+         } catch (IOException e) {
```

Comment:
If I remember correctly, the methods println() and print() from Pretty raise IOException.

But the catch blocks, with System.err.println have no specific reason. I may remove them.